pantheon/src/main/java/tech/pegasys/pantheon/controller/CrossBftPantheonControllerBuilder.java

```diff
@@ -0,0 +1,270 @@
+    return new IbftJsonRpcMethodsFactory(protocolContext);
+  }
+
+  @Override
+  protected SubProtocolConfiguration createSubProtocolConfiguration(
+      final EthProtocolManager ethProtocolManager) {
+    return new SubProtocolConfiguration()
+        .withSubProtocol(Istanbul64Protocol.get(), ethProtocolManager)
+        .withSubProtocol(IbftSubProtocol.get(), new IbftProtocolManager(ibftEventQueue, peers));
+  }
+
+  @Override
+  protected MiningCoordinator createMiningCoordinator(
+      final ProtocolSchedule<IbftContext> protocolSchedule,
+      final ProtocolContext<IbftContext> protocolContext,
+      final TransactionPool transactionPool,
+      final MiningParameters miningParameters,
+      final SyncState syncState,
+      final EthProtocolManager ethProtocolManager) {
+    final MutableBlockchain blockchain = protocolContext.getBlockchain();
+
+    final BlockCreatorFactory blockCreatorFactory =
+        new CrossbftBlockCreatorFactory(
+            (gasLimit) -> gasLimit,
+            transactionPool.getPendingTransactions(),
+            protocolContext,
+            protocolSchedule,
+            miningParameters,
+            nodeKeys);
+
+    final ProposerSelector proposerSelector =
+        new ProposerSelector(blockchain, blockInterface, true);
+
+    // NOTE: peers should not be used for accessing the network as it does not enforce the
+    // "only send once" filter applied by the UniqueMessageMulticaster.
+    final VoteTallyCache voteTallyCache = protocolContext.getConsensusState().getVoteTallyCache();
+    peers = new ValidatorPeers(voteTallyCache);
+
+    final UniqueMessageMulticaster uniqueMessageMulticaster =
+        new UniqueMessageMulticaster(peers, configOptions.getGossipedHistoryLimit());
+
+    final IbftGossip gossiper =
+        new IbftGossip(uniqueMessageMulticaster, LegacyIbftBlockHeaderFunctions.forCommittedSeal());
+
+    final ScheduledExecutorService timerExecutor =
+        newScheduledThreadPool("IbftTimerExecutor", 1, metricsSystem);
+
+    final BlockOperations blockOperations = new CrossBftBlockOperations(nodeKeys);
+
+    final IbftFinalState finalState =
+        new IbftFinalState(
+            voteTallyCache,
+            nodeKeys,
+            Util.publicKeyToAddress(nodeKeys.getPublicKey()),
+            proposerSelector,
+            uniqueMessageMulticaster,
+            new RoundTimer(ibftEventQueue, configOptions.getRequestTimeoutSeconds(), timerExecutor),
+            new BlockTimer(
+                ibftEventQueue, configOptions.getBlockPeriodSeconds(), timerExecutor, clock),
+            blockCreatorFactory,
+            new MessageFactory(nodeKeys),
+            clock);
+
+    final MessageValidatorFactory messageValidatorFactory =
+        new MessageValidatorFactory(
+            proposerSelector,
+            protocolSchedule,
+            protocolContext,
+            (signedPayload, proposedBlock) -> true, blockOperations);
```

Comment:
could still do some validation here obviously not the round number but could probably validate that the proposal payload digest matches the proposed blocks hash